src/xdocs/index.xml.vm

```diff
@@ -217,6 +217,27 @@
       </div>
     </section>
 
+    <section name="Supported Language Level">
+      <p>
+        Checkstyle's grammar and checks currently support all Java language features introduced
+        in Java 21 and below (both regular and preview), except for:
```

Comment:
I would prefer to not mention specific preview features at all, and just say that we may support preview features depending on community demand and that any preview feature support may change at any time.

Reply:
done